src/awkward/highlevel.py

```diff
@@ -1309,6 +1358,20 @@ def _repr(self, limit_cols):
         else:
             valuestr = "-typetracer"
 
+        # prepare named_axis str for repr
+        axisstr = _prettify_named_axes(self.named_axis).split(", ")
+        # we reserve at maximum 20 characters for the named axis string
+        # if it's too long we cut it off until there is enough space for in total 20 chars including ",..."
+        if len(",".join(axisstr)) > 20:
+            while len(",".join(axisstr)) > 20 - len(",..."):
+                axisstr.pop(-1)
+            axisstr.append("...")
+        axisstr = ",".join(axisstr)
+        if axisstr:
+            axisstr = f" {axisstr}"
+        # subtract the reserved space from the limit_cols
+        limit_cols -= len(axisstr)
```

Comment:
I'm mildly in favour of making this elision logic part of `_prettify_named_axes`, so that the implementation details of the string don't leak out to this part of the code.